etl/app/server/s3.js

```diff
@@ -106,6 +107,95 @@ export async function uploadFilePublic(filePath, fileToUpload) {
   }
 }
 
+// Retries an HTTP request in response to a failure
+function retryRequest(serviceName, count, s3Config, callback) {
+  log.info(`Non-200 response returned from ${serviceName} service, retrying`);
+  if (count < s3Config.config.retryLimit) {
+    return setTimeout(
+      () => callback(s3Config, count + 1),
+      s3Config.config.retryIntervalSeconds * 1000,
+    );
+  } else {
+    throw new Error(`Sync ${serviceName} retry count exceeded`);
+  }
+}
+
+// Sync the domain values corresponding to a single domain name
+function fetchSingleDomain(name, mapping) {
+  return async function (s3Config, retryCount = 0) {
+    try {
+      const res = await axios.get(
+        `${s3Config.services.domainValues}?domainName=${mapping.domainName}`,
+      );
+
+      if (res.status !== 200) {
+        return retryRequest(
+          `Domain Values (${mapping.domainName})`,
+          retryCount,
+          s3Config,
+          syncSingleDomain(name, mapping),
+        );
+      }
+
+      const values = res.data.map((value) => {
+        return {
+          label: value[mapping.labelField ?? 'name'],
+          value: value[mapping.valueField ?? 'code'],
```

Comment:
Just a thought. We may want to bring in the other fields from the domains service. For example, we are currently only pulling the `context` value from `OrgStateCode`, but I could see the `name` and `context2` values being useful. Which would help us display something like `PENOBSCOTINDIANNATIONDNR (Maine) (Tribe)`. 

I'm ok with leaving it as is for now and adding in the other fields if the need arises.

Reply:
I think we can pull in the other values as needed, but currently the values are being pulled in so that they can be used as _label, value_ options for inputs.

Reply:
We'd probably pull in the other fields conditionally, right? Then could we lump them into a "description" or our own "context" field for such uses?

Reply:
I think this would be conditionally. Most of the time we would probably be good just pulling in the `name` and `code` fields, but there may be some domains where we want to pull in other fields to assist in building what we display in the UI (like in my `OrgStateCode` example above).